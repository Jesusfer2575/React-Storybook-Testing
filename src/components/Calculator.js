import React, { useState, useEffect } from 'react';
import { typeOperation } from '../utils/constants';

const Calculator = ({ valueA, valueB, operation }) => {
  const [operationState, setOperationState] = useState(operation);
  let sum = 0;
  let multiplication = 0;
  let substraction = 0;
  let division = 0;

  useEffect(() => {
    setOperationState(operation);
  }, [operation]);

  if (valueA && valueB){
    sum = valueA + valueB;
    multiplication = valueA * valueB;
    substraction = valueA - valueB;
    division = valueA / valueB;
  }

  return ( 
    <React.Fragment>
      {(typeOperation.SUM === operationState) && (
        <div>
          The sum of {valueA} and {valueB} is: {sum}
        </div>)}
      
      {(typeOperation.MULTIPLICATION === operationState) && (
        <div>
          The multiplication of {valueA} and {valueB} is: {multiplication}
        </div>)}

      {(typeOperation.SUBSTRACTION === operationState) && (
        <div>
          The substraction of {valueA} and {valueB} is: {substraction}
        </div>)}

      {(typeOperation.DIVISION === operationState) && (
        <div>
          The division of {valueA} and {valueB} is: {division}
        </div>)}
    </React.Fragment>
  );
}
 
export default Calculator;