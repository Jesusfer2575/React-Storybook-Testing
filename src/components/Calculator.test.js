import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';
import { shallow } from 'enzyme';

describe('Calculator', () => {
  it('should render correctly Sum', () => {
    const component = shallow(<Calculator valueA={1} valueB={1} operation={1}/>);
  
    expect(component).toMatchSnapshot();
  });
  it('should render correctly Substraction', () => {
    const component = shallow(<Calculator valueA={1} valueB={1} operation={2}/>);
  
    expect(component).toMatchSnapshot();
  });
  it('should render correctly Multiplication', () => {
    const component = shallow(<Calculator valueA={1} valueB={1} operation={3}/>);
  
    expect(component).toMatchSnapshot();
  });
});