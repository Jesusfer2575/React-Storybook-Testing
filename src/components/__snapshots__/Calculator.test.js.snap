// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render correctly Multiplication 1`] = `
<div>
  <div>
    The multiplication of 
    1
     and 
    1
     is: 
    1
  </div>
</div>
`;

exports[`Calculator should render correctly Substraction 1`] = `
<div>
  <div>
    The substraction of 
    1
     and 
    1
     is: 
    0
  </div>
</div>
`;

exports[`Calculator should render correctly Sum 1`] = `
<div>
  <div>
    The sum of 
    1
     and 
    1
     is: 
    2
  </div>
</div>
`;
